src/dcm/agent/plugins/builtin/unmount_volume.py

```diff
@@ -17,24 +17,57 @@
 import dcm.agent.plugins.api.base as plugin_base
 import dcm.agent.plugins.api.utils as plugin_utils
 import dcm.agent.utils as utils
+import logging
+import os
+import stat
 
+_g_logger = logging.getLogger(__name__)
+
+def _translate_ubuntu_device_id(deviceId):
+    mode = os.stat("/dev/" + deviceId).st_mode
+    if not stat.S_ISBLK(mode) and deviceId.startswith("sd"):
+        return deviceId.replace("sd", "xvd")
+    return deviceId
+
+_device_translate_map = {
+    "ubuntu": _translate_ubuntu_device_id
+}
 
 def _unmount(conf, mount_point):
     command = [conf.get_script_location("unmount"), mount_point]
     (stdout, stderr, rc) = plugin_utils.run_command(conf, command)
     if rc != 0:
+        _g_logger.info("The unmount of %s did not succeed: %s" % (mount_point, stderr))
         raise exceptions.AgentExecutableException(command, rc, stdout, stderr)
     return rc
 
-
-def close_encrypted_device(conf, encrypted_device_id):
+def _close_encrypted_device(conf, encrypted_device_id):
     command = [conf.get_script_location("closeEncryption"),
                encrypted_device_id]
     (stdout, stderr, rc) = plugin_utils.run_command(conf, command)
     if rc != 0:
+        _g_logger.info("The close of encrypted %s did not succeed: %s" % (encrypted_device_id, stderr))
         raise exceptions.AgentExecutableException(command, rc, stdout, stderr)
     return rc
 
+def _check_for_device_translations(platform, deviceId):
+    _g_logger.info("Checking for device translations")
+    try:
+        deviceId = _device_translate_map[platform](deviceId)
+    except KeyError:
+        raise Exception("There is no device translation for %s" % platform)
+    return deviceId
+
+def _find_device_to_unmount(conf, deviceId, platform, device_mappings):
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Rename this parameter "deviceId" to match the regular expression ^[_a-z][a-z0-9_]*$. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.int.enstratius.com/coding_rules#rule_key=python%3AS117)